app/controllers/api/v1/packages_controller.rb

```diff
@@ -322,6 +342,7 @@ def package_record
         else
           @package.assign_attributes(package_params)
         end
+        @package.detail = assign_detail if params["package"]["detail_type"].present?
```

Comment:
yeah @steveyken I was trying to get it work without this method but it seems that `accept_nested_attributes_for` works for has_one/has_many relationship. `detail` is in `belongs_to` association with package, because detail is polymorphic.

one workaround for this can be -> https://stackoverflow.com/a/3975758
other option is the way I have done it in the controller itself.

cc- @namrataukirde